From d983452a22ce867748cda2434b54a464891a4e20 Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@remirepo.net>
Date: Wed, 15 Apr 2020 10:58:40 +0200
Subject: [PATCH] Fix MATCHES usage so ENABLE_STATIC=BUILD_ONLY works

---
 CMakeLists.txt | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9705540a8..cede9636c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -152,7 +152,7 @@ if (NOT ENABLE_BSON MATCHES "ON|AUTO|SYSTEM")
    message (FATAL_ERROR "ENABLE_BSON option must be ON, AUTO, or SYSTEM")
 endif ()
 
-if (NOT ENABLE_STATIC MATCHES "ON|OFF|AUTO|BUILD_ONLY|DONT_INSTALL")
+if (NOT ENABLE_STATIC MATCHES "^(ON|OFF|AUTO|BUILD_ONLY|DONT_INSTALL)$")
    message (FATAL_ERROR
       "ENABLE_STATIC option must be ON, OFF, AUTO, BUILD_ONLY, or DONT_INSTALL"
    )
@@ -161,11 +161,15 @@ endif ()
 set (MONGOC_ENABLE_STATIC_BUILD 0)
 set (MONGOC_ENABLE_STATIC_INSTALL 0)
 
-if (ENABLE_STATIC MATCHES "ON|AUTO")
+if (ENABLE_STATIC MATCHES "^(ON|AUTO)$")
+   message (STATUS "Build and install static libraries")
    set (MONGOC_ENABLE_STATIC_BUILD 1)
    set (MONGOC_ENABLE_STATIC_INSTALL 1)
-elseif (ENABLE_STATIC MATCHES "BUILD_ONLY|DONT_INSTALL")
+elseif (ENABLE_STATIC MATCHES "^(BUILD_ONLY|DONT_INSTALL)$")
+   message (STATUS "Build only static libraries")
    set (MONGOC_ENABLE_STATIC_BUILD 1)
+else ()
+   message (STATUS "Don't build static libraries")
 endif()
 
 if (ENABLE_BSON STREQUAL SYSTEM)
