diff --git a/doc/taglist.py b/doc/taglist.py
index 9283b6f..954d050 100755
--- a/doc/taglist.py
+++ b/doc/taglist.py
@@ -25,7 +25,10 @@ THE SOFTWARE.
 
 from docutils import nodes, utils
 from sphinx.environment import NoUri
-from sphinx.util.compat import Directive
+try:
+    from sphinx.util.compat import Directive
+except ImportError:
+    from docutils.parsers.rst import Directive
 
 
 def get_tags(s):
