From a021ada41374b9be9b478252d53a2a199bfdbf7b Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Fri, 14 Oct 2016 09:42:13 +0200
Subject: [PATCH] CDRIVER-1702 don't fail when returned platform is truncated

---
 tests/test-mongoc-handshake.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tests/test-mongoc-handshake.c b/tests/test-mongoc-handshake.c
index b806b59..604cbf8 100644
--- a/tests/test-mongoc-handshake.c
+++ b/tests/test-mongoc-handshake.c
@@ -257,7 +257,9 @@ test_mongoc_handshake_data_append_success (void)
    ASSERT (BSON_ITER_HOLDS_UTF8 (&md_iter));
    val = bson_iter_utf8 (&md_iter, NULL);
    ASSERT (val);
-   ASSERT (strstr (val, platform) != NULL);
+   if (strlen(val) < 250) { /* standard val are < 100, may be truncated on some platform */
+      ASSERT (strstr (val, platform) != NULL);
+   }
 
    mock_server_replies_simple (request,
                                "{'ok': 1, 'ismaster': true}");
