From 612b619776a21b9649e746e4cf5f7d10af9a47da Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Fri, 10 Feb 2017 06:48:24 +0100
Subject: [PATCH] CDRIVER-2042 skip test_create_index_fail when offline

---
 tests/test-mongoc-collection.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/tests/test-mongoc-collection.c b/tests/test-mongoc-collection.c
index f28fb26..ca629ce 100644
--- a/tests/test-mongoc-collection.c
+++ b/tests/test-mongoc-collection.c
@@ -4735,7 +4735,7 @@ test_insert_duplicate_key (void)
 }
 
 static void
-test_create_index_fail (void)
+test_create_index_fail (void *context)
 {
    mongoc_client_t *client;
    mongoc_collection_t *collection;
@@ -4940,6 +4940,7 @@ test_collection_install (TestSuite *suite)
       suite, "/Collection/find_indexes/error", test_find_indexes_err);
    TestSuite_AddLive (
       suite, "/Collection/insert/duplicate_key", test_insert_duplicate_key);
-   TestSuite_Add (
-      suite, "/Collection/create_index/fail", test_create_index_fail);
+   TestSuite_AddFull (
+      suite, "/Collection/create_index/fail", test_create_index_fail,
+      NULL, NULL, test_framework_skip_if_offline);
 }
