From 58afb1fd3c59fcc59021227d8f2b490275fd3105 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Wed, 11 Jan 2017 09:47:33 +0100
Subject: [PATCH] fix test for ipv6

---
 tests/test-mongoc-uri.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/test-mongoc-uri.c b/tests/test-mongoc-uri.c
index 223a8c4..169c817 100644
--- a/tests/test-mongoc-uri.c
+++ b/tests/test-mongoc-uri.c
@@ -543,7 +543,7 @@ test_mongoc_host_list_from_string (void)
    ASSERT(!strcmp(host_list.host_and_port, "localhost:27019"));
    ASSERT(!strcmp(host_list.host, "localhost"));
    ASSERT(host_list.port == 27019);
-   ASSERT(host_list.family == AF_INET);
+   ASSERT(host_list.family == AF_INET || host_list.family == AF_UNSPEC);
    ASSERT(!host_list.next);
 }
 
